invenio_records_resources/services/files/components/metadata.py

```diff
@@ -20,8 +21,9 @@ class FileMetadataComponent(FileServiceComponent):
 
     def init_files(self, id, identity, record, data):
         """Init files handler."""
-        # TODO: Load via marshmallow schema?
-        for file_metadata in data:
+        schema = InitFileSchema(many=True)
```

Comment:
comment: ha the `many=True` tricked me. I was thinking how a list can pass the loading step if there is no `Nested`/`List` in the schema. Nice!

Reply:
Yeah it's a nice trick!